crates/rccheck/src/lib.rs

```diff
@@ -473,6 +485,48 @@ fn prepare_for_self_signed<'a>(
     Ok((cert, intermediates, vec![root]))
 }
 
+// This prepares arguments for webpki, including a trust anchor which is the end entity of the certificate
+// (which embodies a self-signed certificate by definition)
+pub fn prepare_for_self_signed_test<'a>(
+    end_entity: &'a rustls::Certificate,
+    intermediates: &'a [rustls::Certificate],
+    root_spki: &'a Vec<u8>
+) -> Result<CertChainAndRoots<'a>, rustls::Error> {
+    // EE cert must appear first.
+    let cert = webpki::EndEntityCert::try_from(end_entity.0.as_ref()).map_err(pki_error)?;
+
+    let intermediates: Vec<&'a [u8]> = intermediates.iter().map(|cert| cert.0.as_ref()).collect();
+
+    // reinterpret the certificate as a root, materializing the self-signed policy
+    let root_anchor = webpki::TrustAnchor {
+        subject: &[],
+        spki: &root_spki[2..],
```

Comment:
Seems... unsafe - but I couldn't find any other interface that allows us to do this without a private key of the spki.

Reply:
So, as you noted the fields of this struct are public. This is actually fine (x509 data structures are mostly fitting parsing needs, rather than security functions).

What I don't get is why you need the `[2..]`.

Reply:
I think there is a difference in representation across libraries. One library prefixes with two extra bytes (one byte for the length, another is the first byte of the spki)